### PR TITLE
Correct SPDX-License-Identifier

### PR DESCRIPTION
**Pull request by pshipton**
Issue https://github.com/eclipse-openj9/openj9/issues/17429

### CODE REVIEW

**Comment by keithc-ca** [comment]
`NOTICE.md` should be updated as well. I suggest the few lines preceding the SPDX expression should be improved; there's a mix of SPDX terms and plain English.

**Comment by pshipton** [comment]
Updated NOTICE.md. I've copied the preceding lines from the template in https://www.eclipse.org/projects/tools/documentation.php?id=technology.openj9

File: NOTICE.md [hunk 1]
```diff
@@ -22,14 +22,12 @@ https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0 which
...
-'https://www.gnu.org/software/classpath/license.html.
-
-SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH
-OpenJDK-assembly-exception-1.0 OR GPL-2.0-with-classpath-exception
+are satisfied: GPL-2.0 with Classpath-exception-2.0 which is available at
+https://openjdk.java.net/legal/gplv2+ce.html, and GPL-2.0 with Assembly
+Exception which is available at
+https://openjdk.java.net/legal/assembly-exception.html.
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only with Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
```
Comment (keithc-ca) [inline NOTICE.md:30]: This should use `WITH` (instead of `with`).

The eclipse template should be corrected as well (I see multiple such errors there).

Comment (keithc-ca) [reply NOTICE.md:30]: The license expression for doxygen is also incorrect.

Comment (pshipton) [reply NOTICE.md]: Updated

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**